File: zenoh_elixir/lib/zenoh_elixir/sub.ex
defmodule ZenohElixir.Sub do
  def main do
    {:ok, session} = Zenohex.open()
    {:ok, subscriber} = Zenohex.Session.declare_subscriber(session, "key/expression")

    spawn(ZenohElixir.Sub, :subscribe, [subscriber])
  end

  def subscribe(subscriber) do
    case Zenohex.Subscriber.recv_timeout(subscriber) do
      {:error, :timeout} -> nil
      {:ok, msg} -> IO.puts "[sub.ex] " <> msg.value
    end

    subscribe(subscriber)
  end
end

File: zenoh_native/src/bin/sub.rs
use zenoh::prelude::r#async::*;

#[async_std::main]
async fn main() {
    let session = zenoh::open(config::default()).res().await.unwrap();
    let subscriber = session.declare_subscriber("key/expression").res().await.unwrap();
    while let Ok(msg) = subscriber.recv_async().await {
        println!("[sub.rs] {}", msg.value);
    };
}
